msm/src/fec/mod.rs

```diff
@@ -83,6 +83,56 @@ mod tests {
         build_fec_addition_circuit(&mut rng, 1 << 4);
     }
 
+    #[test]
+    pub fn test_regression_relation_constraints_fec() {
+        let mut constraint_env = ConstraintBuilderEnv::<Fp, LookupTable<Ff1>>::create();
+        constrain_ec_addition::<Fp, Ff1, _>(&mut constraint_env);
+        let constraints = constraint_env.get_relation_constraints();
+
+        let mut constraints_degrees = HashMap::new();
+
+        assert_eq!(constraints.len(), 24);
+
+        {
+            constraints.iter().for_each(|c| {
+                let degree = c.degree(1, 0);
+                *constraints_degrees.entry(degree).or_insert(0) += 1;
+            });
+
+            assert_eq!(constraints_degrees.get(&1), None);
+            assert_eq!(constraints_degrees.get(&2), Some(&3));
+            assert_eq!(constraints_degrees.get(&3), Some(&21));
+
+            assert!(constraints.iter().map(|c| c.degree(1, 0)).max() <= Some(3));
+        }
+    }
+
+    #[test]
+    pub fn test_regression_constraints_fec() {
+        let mut constraint_env = ConstraintBuilderEnv::<Fp, LookupTable<Ff1>>::create();
+        constrain_ec_addition::<Fp, Ff1, _>(&mut constraint_env);
+        let constraints = constraint_env.get_constraints();
+
+        let mut constraints_degrees = HashMap::new();
+
+        assert_eq!(constraints.len(), 63);
+
+        {
+            constraints.iter().for_each(|c| {
+                let degree = c.degree(1, 0);
+                *constraints_degrees.entry(degree).or_insert(0) += 1;
+            });
+
+            assert_eq!(constraints_degrees.get(&1), Some(&1));
+            assert_eq!(constraints_degrees.get(&2), Some(&5));
+            assert_eq!(constraints_degrees.get(&3), Some(&21));
+            assert_eq!(constraints_degrees.get(&4), None);
+            assert_eq!(constraints_degrees.get(&5), Some(&2));
+            assert_eq!(constraints_degrees.get(&6), None);
+            assert_eq!(constraints_degrees.get(&7), Some(&34));
```

Comment:
Degree 7 here is because of the lookups I assume.

Reply:
Yes.